File: src/create/builder/pizzaj/Application.java
package create.builder.pizzaj;

/**
 * @author Ting
 * @date 2020/8/7
 * @description
 */
public class Application {
    public static void main(String[] args) {
        PizzaBuilderDirector director = new PizzaBuilderDirector();
        Pizza pizza = director.constructor(new FruitPizzaBuilder());
        System.out.println(pizza.toString());
    }
}

File: src/create/abstract_factory/button/Application.kt
package create.abstract_factory.button

import create.abstract_factory.button.button.Button
import create.abstract_factory.button.checkbox.CheckBox
import create.abstract_factory.button.factory.GUIFactory

/**
 * @author Ting
 * @date 2020/7/31
 * @description
 */
class Application(factory: GUIFactory) {
    var mButton: Button = factory.createButton()
    var mCheckBox: CheckBox = factory.createCheckbox()

    fun paint() {
        mButton.paint()
        mCheckBox.paint()
    }
}